test/TaskQueue_test.js

```diff
@@ -0,0 +1,99 @@
+const assert = require('assert');
+const debug = require('debug')('index:test:api_test');
+const helper = require('./helper');
+const slugid = require('slugid');
+const taskcluster = require('taskcluster-client');
+const request = require('superagent');
+const assume = require('assume');
+const libUrls = require('taskcluster-lib-urls');
+
+helper.secrets.mockSuite('TaskQueue_test.js', ['taskcluster'], function(mock, skipping) {
+  helper.withFakeQueue(mock, skipping);
+
+  test('Run task and test indexing', async function() {
+    const tq = await helper.load('taskqueue');
+    helper.claimableWork.push({
+      tasks: [
+        {
+          status: {
+            taskId: 0,
+          },
+          runId: 0,
+          workerGroup: 'garbage-hybrid1999',
+          workerId: 'succeed',
+          task: {
+            provisionerId: 'garbage-hybrid1999',
+            workerType: 'succeed',
+            payload: {},
+          },
+        },
+      ],
+    });
+    await tq.claimTasks();
+    assert.deepEqual(helper.taskResolutions[0], {completed: true});
+    helper.claimableWork.push({
```

Comment:
This is a good place to break this into multiple tests -- one each for `{completed: true}`, `{failed: true}`, and `{exception: ..}`.  Once you've broken this into three tests, it might be good to factor out a bit of the common code, perhaps into a `makeClaimableWork(payload)` function.